pages/learn/deploy/offline-updates.md

```diff
@@ -0,0 +1,271 @@
+
+```bash
+$ uuid=$(uuidgen | tr '[:upper:]' '[:lower:]' | sed 's/-//g')
+```
+
+With `balena device register`, devices can be preregistered to a balenaCloud application involving a simple call with a unique identifier for the device. You can read more about the full process of pre-registering a device in the [balena-cli advanced masterclass][balena-cli advanced masterclass]. This step can be skipped if a pre-existing device is needed to be updated.
+
+```bash
+$ balena device register ${app_slug} --uuid ${uuid}
+```
+
+### Download balenaOS Image
+
+Download the latest production version of [balenaOS][balenaOS] for the device type initialized in the `device_type` version.
+
+```bash
+$ os_version=$(balena os versions ${device_type} | grep prod | head -n 1 | awk '{print $1}')
+$ tmpimg=$(mktemp).img
+
+$ balena os download ${device_type} \
+  --output ${tmpimg} \
+  --version ${os_version}
+```
+
+### Configure balenaOS Image
+
+Next, configure the downloaded image by injecting a [config.json][config-file] file using the [`balena os configure`][balena-os-configure] command. Most common system settings can be (re)specified via [config.json](config-file). To preserve the pre-existing registered device's identity, the same `uuid` initialized earlier will be used to generate a config.json file. 
+
+```bash
+$ tmpconfig=$(mktemp)
+$ config=$(mktemp)
+
+$ balena config generate \
+    --device ${uuid} \
+    --version ${os_version} \
+    --network ethernet \
+    --appUpdatePollInterval 10 \
+    --output ${tmpconfig} \
+    && cat ${tmpconfig} | jq .
+
+$ cat < ${tmpconfig} \
+    | jq --arg keys "$(cat ${ssh_key})" '. + {os: {sshKeys: [$keys]}}' > "${config}" \
+    && cat < ${config} | jq .
+
+$ balena os configure ${tmpimg} \
+    --app ${app_slug} \
+    --device-type ${device_type} \
+    --version ${os_version} \
+    --config-network ethernet \
+    --config ${config}
+
+$ rm ${tmpconfig}
+$ rm ${config}
+```
+
+### Create and Preload Release
+
+Offline update revolve around the concept of [balena preload][balena-preload]. Preload is used to flash the balenaOS image and your application release in a single step, so the device starts running your application containers as soon as it boots. Preloading removes the need for your devices to download the initial application images directly from balena's build servers, making it an ideal base for the offline update process. Read more about [preloading a device image][preloading-device-image].
+
+> Important note: [balena preload](https://github.com/balena-io/balena-cli/blob/master/INSTALL-MAC.md#balena-preload) functionality requires Docker with AUFS support.
+
+Since preload involves flashing an application release with the balenaOS image, if the pre-existing application doesn't have any releases, then a release needs to be created. If present already, then the next step can be skipped. 
+
+```bash
+$ date +%s > app/.epoch
+
+$ balena deploy ${app_slug} --build --source .
```

Comment:
@vipulgupta2048 @thgreasi we do need to add `--emulated` flag is these instructions are running under Linux (Ubuntu 20.04 LTS). I tested today and without emulated, I get an exec error during the build/deploy.

Reply:
@ab77 Cool, Adding an emulated flag until someone says otherwise.